infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -68,9 +70,10 @@ public synchronized void handleMessageSetEpoch(@NonNull CorfuPayloadMsg<Long> ms
                                                    ChannelHandlerContext ctx,
                                                    @NonNull IServerRouter r) {
         try {
-            log.info("handleMessageSetEpoch: Received SET_EPOCH, moving to new epoch {}",
-                    msg.getPayload());
-            serverContext.setServerEpoch(msg.getPayload(), r);
+            long epoch = msg.getPayload();
+            log.info("handleMessageSetEpoch: Received SET_EPOCH, moving to new epoch {}", epoch);
+            serverContext.setServerEpoch(epoch, r);
+            serverContext.getServers().forEach(s -> s.sealServerWithEpoch(epoch));
```

Comment:
Why didn't we push this logic into setServerEpoch (instead of doing it in the handler) ?